### PR TITLE
Add webhook custom headers

### PR DESCRIPTION
**Pull request by andrelias**
- Add webhook custom headers documentation section
- Add offerId filter on transaction qualified webhooks
- Fix inconsistencies and formatting

### CODE REVIEW

File: webhooks/README.md [hunk 1]
```diff
@@ -1,30 +1,30 @@
 # Webhooks
+
 Fidel uses webhooks to notify your application when relevant events happen in your account. This functionality can be used to receive data from events not triggered by direct API requests or to receive the data in a service that is not responsible for making the API request but needs to consume the response.
 
-We will notify your registered webhook URLs as the event happens. For example, when a customer makes a payment with a linked Mastercard on a participating location, a `transaction.auth` event is sent in real-time to the specified webhook URL with the transactional data in the request payload.
+Fidel will notify your registered webhook URLs as the event happens. For example, when a customer makes a payment with a linked Mastercard on a participating location, a `transaction.auth` event is sent in real-time to the specified webhook URL with the transaction object in the request payload.
+---
```
Comment (c0derabbit) [inline webhooks/README.md:6]: let's add a newline before the `---`, otherwise it may be parsed as a heading instead of a paragraph and a `hr`

File: webhooks/README.md [hunk 2]
```diff
@@ -1,30 +1,30 @@
...
+A unique secret key is generated for each webhook. The key is returned in the response's `secretKey` property if you are using the Webhooks API. You can also copy the key from the dashboard's webhooks page by clicking in the **Show Key** button next to your webhook endpoint. To verify a webhook request, generate a signature using the same key that Fidel uses and compare that to the value of the `x-fidel-signature` header:
 
-1. Create a string concatenating the body of the request, the webhook URL and the timestamp value from the `x-fidel-timestamp` header.
-2. Double hash the resulting string using the webhook key with HMAC-SHA256 and encode it in Base-64.
-3. Compare the signature you generated with the signature provided in the `x-fidel-signature` header.
+> **1.** Create a string concatenating the body of the request, the webhook URL and the timestamp value from the `x-fidel-timestamp` header.
+> **2.** Double hash the resulting string using the webhook key with HMAC-SHA256 and encode it in Base-64.
+> **3.** Compare the signature you generated with the signature provided in the `x-fidel-signature` header.
```
Comment (c0derabbit) [inline webhooks/README.md:27]: let's keep an eye out for how this renders

File: webhooks/README.md [hunk 3]
```diff
@@ -1,30 +1,30 @@
...
+You can create up to five webhook URLs for the same event in the same Program. Fidel sends the data via HTTP POST in JSON format and will send test or live events depending on your dashboard's environment setting or if you are using test or live API keys.
 
 Fidel only accepts HTTPS URLs for webhooks endpoints. In order to create webhooks and receive event data, your server must be configured to support HTTPS with a valid certificate.
 
 <div class="info-box">
-    <small>Important note</small><br/>
+    <small>Return a 200 status code</small><br/>
 	To confirm receipt of a webhook event, your server endpoint should return a <code>200</code> HTTP status code. Any other response will be treated as a failure and we retry the request three times over the next hour with exponential back off.
 </div>
+---
```
Comment (c0derabbit) [inline webhooks/README.md:18]: pls add a newline before

File: webhooks/README.md [hunk 4]
```diff
@@ -53,14 +53,84 @@ function isSignatureValid(fidelHeaders, payload, secret, url) {
 ```
 
 To prevent replay attacks where a valid payload and its signature is intercepted and re-transmitted, you can use the `x-fidel-timestamp` header and confirm that the timestamp is not too old. We recommend you validate the requests in a 5 minute gap. In case of retries, a new signature and timestamp are generated for each new request.
+---
```
Comment (c0derabbit) [inline webhooks/README.md:56]: newline before please

File: webhooks/README.md [hunk 5]
```diff
@@ -53,14 +53,84 @@ function isSignatureValid(fidelHeaders, payload, secret, url) {
...
+
+## Custom headers
+Fidel allows you to define custom HTTP headers that are passed in the webhook POST requests sent to you application.
+Custom headers can be defined when creating a new webhook in the dashboard or by using the [Webhooks API](https://reference.fidel.uk/v1/reference#create-webhook-brand) by setting the optional `headers` object with a key-value pair. You can use the [Update  Webhook](https://reference.fidel.uk/reference#create-webhook-program) endpoint and submit an empty `headers` object in order to remove the custom headers from the webhook.
```
Comment (c0derabbit) [inline webhooks/README.md:60]: let's add a newline between these paragraphs so they render separately

File: webhooks/README.md [hunk 6]
```diff
@@ -53,14 +53,84 @@ function isSignatureValid(fidelHeaders, payload, secret, url) {
...
+```sh
+curl -X POST \
+  https://api.fidel.uk/v1/programs/06471dbe-a3c7-429e-8a18-16dc97e5cf35/hooks \
+  -H 'Content-Type: application/json' \
+  -H 'Fidel-Key: sk_test_50ea90b6-2a3b-4a56-814d-1bc592ba4d63' \
+  -d '{
+        "event": "transaction.auth",
+        "url": "https://realtime.com",
+        "headers": {
+  	       "Custom-Header": "my-custom-header"
```
Comment (c0derabbit) [inline webhooks/README.md:73]: let's change these 6 spaces to 2, to keep consistent formatting

File: webhooks/README.md (hunk 2 above)
Comment (andrelias) [reply webhooks/README.md:27]: Yes :) This is an experiment. Let's see how it looks.
Do we still have an develop environment setup. Would be good if we could deploy first to dev to see the changes before merging to master.

File: webhooks/README.md [hunk 7]
```diff
@@ -53,14 +53,84 @@ function isSignatureValid(fidelHeaders, payload, secret, url) {
...
+  "Fidel-User"
+]
+```
 ---
 
+
 ## Events
 
 ### Brand
-
```
Comment (c0derabbit) [inline webhooks/README.md:62]: removing these newlines will do nothing on the frontend, we'll just have to fix margins and we already have a card for that – but they do no harm, just fyi.

File: webhooks/README.md (hunk 5 above)
Comment (goncaloneves) [inline webhooks/README.md:60]: `You can use the [Update  Webhook](https://reference.fidel.uk/reference#create-webhook-program) endpoint and submit an empty `headers` object in order to remove the custom headers from the webhook.`

1. Should we not link to all endpoints that allow headers? There are three.
2. In this block we use as example `https://reference.fidel.uk/v1/reference#create-webhook-brand` and on L62 we use `https://reference.fidel.uk/reference#create-webhook-program`.
3. I would reword this without specifying the PUT, otherwise it might seem it's the only place that you can send empty headers. Something like: `You can reset (or remove) your custom headers from the webhook by submitting an empty headers object.`

File: webhooks/README.md (hunk 2 above)
Comment (c0derabbit) [reply webhooks/README.md:27]: not now, we're moving to a different flow. shouldn't be a problem, I'd expect it to
> **1.** render as a callout
> **2.** like this

i'm confident enough, but here to fix if it breaks

File: webhooks/README.md [hunk 8]
```diff
@@ -53,14 +53,84 @@ function isSignatureValid(fidelHeaders, payload, secret, url) {
...
+  "Upgrade",
+  "Via",
+  "Warning",
+  "Fidel-Account",
+  "Fidel-Key",
+  "Fidel-Live",
+  "Fidel-Request-Id",
+  "X-Fidel-Signature",
+  "X-Fidel-Timestamp",
+  "Fidel-User"
```
Comment (goncaloneves) [inline webhooks/README.md:124]: Order this guy alphabetically.

File: webhooks/README.md (hunk 6 above)
Comment (goncaloneves) [inline webhooks/README.md:73]: Should we not add `Authorization` as one of the example headers? It is the most commonly used Authentication header in http requests. Same applies for API reference.

Comment (c0derabbit) [reply webhooks/README.md:73]: don't know if that's necessary though or communicates well enough that it can be `any-key-you-want` (except for the blacklisted ones)?

File: webhooks/README.md (hunk 5 above)
Comment (c0derabbit) [reply webhooks/README.md:60]: to 3., would you use any other method than `PUT` to update a webhook?

Comment (goncaloneves) [reply webhooks/README.md:60]: You can send POST with empty headers and it will be accepted.

Comment (andrelias) [reply webhooks/README.md:60]: I updated the link of the first to the create hook by program and the second to the PUT.

File: webhooks/README.md (hunk 6 above)
Comment (goncaloneves) [reply webhooks/README.md:73]: What I think it is missing is the what is the use-case for Custom Headers?
It is mostly used for Authentication on the listening server, so we should probably add a block explaining that and `Authorization` header the most commonly used one.

Comment (andrelias) [reply webhooks/README.md:73]: Agree with Eszter. Should be clear that is custom.
